### ISSUE TITLE
Database: error on upgrade

### ISSUE THREAD

**Issue by tobiasKaminsky** [nextcloud/android#1341]
```
 MultiDex: VM has multidex support, MultiDex support library is disabled.
08-14 10:39:09.423 13815 13815 I FirebaseInitProvider: FirebaseApp initialization unsuccessful
08-14 10:39:09.450 13815 13815 I FA      : App measurement is starting up, version: 10240
08-14 10:39:09.450 13815 13815 I FA      : To enable debug logging run: adb shell setprop log.tag.FA VERBOSE
08-14 10:39:09.455 13815 13815 E FA      : GoogleService failed to initialize, status: 10, Missing google app id value from from string resources with name google_app_id.
08-14 10:39:09.456 13815 13815 E FA      : Scheduler not set. Not logging error/warn
08-14 10:39:09.456 13815 13815 I FA      : To enable faster debug mode event logging run:
08-14 10:39:09.456 13815 13815 I FA      :   adb shell setprop debug.firebase.analytics.app com.nextcloud.client
08-14 10:39:09.480 13815 13831 E FA      : Missing google_app_id. Firebase Analytics disabled. See https://goo.gl/NAOOOI
08-14 10:39:09.484 13815 13815 I JobManager: Found pending job request{id=2019, tag=FilesSyncJob}, canceling
08-14 10:39:09.526 13815 13833 D ThumbnailsCacheManager: create dir: /storage/emulated/0/Android/data/com.nextcloud.client/cache/thumbnailCache
08-14 10:39:09.532 13815 13815 D JobProxy24: Schedule periodic jobInfo success, request{id=2020, tag=FilesSyncJob}, interval 00:15:00, flex 00:05:00
08-14 10:39:09.538 13815 13815 I SQL     : Entering in onUpgrade
08-14 10:39:09.549 13815 13815 I SQL     : OUT of the ADD in onUpgrade; oldVersion == 22, newVersion == 23
08-14 10:39:09.549 13815 13815 I SQL     : OUT of the ADD in onUpgrade; oldVersion == 22, newVersion == 23
08-14 10:39:09.550 13815 13815 I SQL     : OUT of the ADD in onUpgrade; oldVersion == 22, newVersion == 23
08-14 10:39:09.550 13815 13815 I SQL     : OUT of the ADD in onUpgrade; oldVersion == 22, newVersion == 23
08-14 10:39:09.550 13815 13815 I SQL     : OUT of the ADD in onUpgrade; oldVersion == 22, newVersion == 23
08-14 10:39:09.550 13815 13815 I SQL     : OUT of the ADD in onUpgrade; oldVersion == 22, newVersion == 23
08-14 10:39:09.550 13815 13815 I SQL     : OUT of the ADD in onUpgrade; oldVersion == 22, newVersion == 23
08-14 10:39:09.550 13815 13815 I SQL     : OUT of the ADD in onUpgrade; oldVersion == 22, newVersion == 23
08-14 10:39:09.551 13815 13815 I SQL     : OUT of the ADD in onUpgrade; oldVersion == 22, newVersion == 23
08-14 10:39:09.551 13815 13815 I SQL     : OUT of the ADD in onUpgrade; oldVersion == 22, newVersion == 23
08-14 10:39:09.551 13815 13815 I SQL     : OUT of the ADD in onUpgrade; oldVersion == 22, newVersion == 23
08-14 10:39:09.551 13815 13815 I SQL     : OUT of the ADD in onUpgrade; oldVersion == 22, newVersion == 23
08-14 10:39:09.551 13815 13815 I SQL     : OUT of the ADD in onUpgrade; oldVersion == 22, newVersion == 23
08-14 10:39:09.551 13815 13815 I SQL     : OUT of the ADD in onUpgrade; oldVersion == 22, newVersion == 23
08-14 10:39:09.551 13815 13815 I SQL     : OUT of the ADD in onUpgrade; oldVersion == 22, newVersion == 23
08-14 10:39:09.552 13815 13815 I SQL     : OUT of the ADD in onUpgrade; oldVersion == 22, newVersion == 23
08-14 10:39:09.552 13815 13815 I SQL     : Entering in the #23 adding type column for synced folders, Create filesystem table
08-14 10:39:09.552 13815 13815 I SQL     : Add type column and default value 0 (CUSTOM) to synced_folders table
08-14 10:39:09.552 13815 13815 E SQLiteLog: (1) duplicate column name: type
08-14 10:39:09.554 13815 13815 E FileContentProvider: ERROR!
08-14 10:39:09.554 13815 13815 E FileContentProvider: android.database.sqlite.SQLiteException: duplicate column name: type (code 1): , while compiling: ALTER TABLE synced_folders ADD COLUMN type INTEGER  DEFAULT 0
08-14 10:39:09.554 13815 13815 E FileContentProvider: 	at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
08-14 10:39:09.554 13815 13815 E FileContentProvider: 	at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:889)
08-14 10:39:09.554 13815 13815 E FileContentProvider: 	at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:500)
08-14 10:39:09.554 13815 13815 E FileContentProvider: 	at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:588)
08-14 10:39:09.554 13815 13815 E FileContentProvider: 	at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:58)
08-14 10:39:09.554 13815 13815 E FileContentProvider: 	at android.database.sqlite.SQLiteStatement.<init>(SQLiteStatement.java:31)
08-14 10:39:09.554 13815 13815 E FileContentProvider: 	at android.database.sqlite.SQLiteDatabase.executeSql(SQLiteDatabase.java:1677)
08-14 10:39:09.554 13815 13815 E FileContentProvider: 	at android.database.sqlite.SQLiteDatabase.execSQL(SQLiteDatabase.java:1608)
08-14 10:39:09.554 13815 13815 E FileContentProvider: 	at com.owncloud.android.providers.FileContentProvider$DataBaseHelper.onUpgrade(FileContentProvider.java:1071)
08-14 10:39:09.554 13815 13815 E FileContentProvider: 	at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:256)
08-14 10:39:09.554 13815 13815 E FileContentProvider: 	at android.database.sqlite.SQLiteOpenHelper.getReadableDatabase(SQLiteOpenHelper.java:187)
08-14 10:39:09.554 13815 13815 E FileContentProvider: 	at com.owncloud.android.providers.FileContentProvider.query(FileContentProvider.java:449)
08-14 10:39:09.554 13815 13815 E FileContentProvider: 	at android.content.ContentProvider.query(ContentProvider.java:1020)
08-14 10:39:09.554 13815 13815 E FileContentProvider: 	at android.content.ContentProvider$Transport.query(ContentProvider.java:239)
08-14 10:39:09.554 13815 13815 E FileContentProvider: 	at android.content.ContentResolver.query(ContentResolver.java:532)
08-14 10:39:09.554 13815 13815 E FileContentProvider: 	at android.content.ContentResolver.query(ContentResolver.java:474)
```

> android.database.sqlite.SQLiteException: duplicate column name: type (code 1): , while compiling: ALTER TABLE synced_folders ADD COLUMN type INTEGER  DEFAULT 0

From logcat by @jasonbayton 
@jasonbayton you said that you did a fresh install? Weird...

There are only two occasions where we modify the table, one on upgrade (which fails) and one on a fresh installation...

**Comment by AndyScherzinger** [comment]
> There are only two occasions where we modify the table, one on upgrade (which fails) and one on a fresh installation...

@tobiasKaminsky and that is the error!!! the new column has been added to the inital table creation part (which originally has been release with 1.4.x) and to the update part (where it should be, only) for 2.0.0!!! As for the RC I only tested the upgrade not a fresh install. Do you have the time to test this in an emulator?

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky If you don't find the time I might be able to provide a fix tonight.

**Comment by tobiasKaminsky** [comment]
On my emulator I am only doing fresh installations and there it works.

The problem is that the column is also added in db scheme 16 (with createSyncedFoldersTable).
So if you upgrade from 15 to 23 (newest):
- 16 -> add type column
- ...
- 23 -> add type column

(same for wifi and charging column).

**Comment by jasonbayton** [comment]
Yes, new install tried several times now

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky 

16 --> 1.4.3
23 --> 2.0.0

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky this is what 1.4.3 originally did and where the upgrade v23 should be based on:
```
private void createSyncedFoldersTable(SQLiteDatabase db) {
        db.execSQL("CREATE TABLE " + ProviderTableMeta.SYNCED_FOLDERS_TABLE_NAME + "("
                + ProviderTableMeta._ID + " INTEGER PRIMARY KEY, "                          // id
                + ProviderTableMeta.SYNCED_FOLDER_LOCAL_PATH + " TEXT, "           // local path
                + ProviderTableMeta.SYNCED_FOLDER_REMOTE_PATH + " TEXT, "           // remote path
                + ProviderTableMeta.SYNCED_FOLDER_WIFI_ONLY + " INTEGER, "          // wifi_only
                + ProviderTableMeta.SYNCED_FOLDER_CHARGING_ONLY + " INTEGER, "      // charging only
                + ProviderTableMeta.SYNCED_FOLDER_ENABLED + " INTEGER, "            // enabled
                + ProviderTableMeta.SYNCED_FOLDER_SUBFOLDER_BY_DATE + " INTEGER, "  // subfolder by date
                + ProviderTableMeta.SYNCED_FOLDER_ACCOUNT + "  TEXT, "              // account
                + ProviderTableMeta.SYNCED_FOLDER_UPLOAD_ACTION + " INTEGER );"     // upload action
        );
    }
```

**Comment by jasonbayton** [comment]
> There still must be some leftovers

Where else than /sdcard/Android/data and /internalstorage/Android/data should I check for leftovers please?

**Comment by jasonbayton** [comment]
FYI I've again completely removed the client, gone through via a file search against anything owncloud or nextcloud and deleted it, then reinstalled. PM'd the logcat again via help.nc

**Comment by AndyScherzinger** [comment]
Fix has been merged for 2.0.0RC3
